### PR TITLE
fix(components/datetime): notify date range value changes when switching between calculators without datepickers

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
[AB#3079050](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3079050)

### CODE REVIEW

File: libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts [hunk 1]
```diff
@@ -551,13 +545,13 @@ export class SkyDateRangePickerComponent
...
 
-    const valueOrDefault =
-      !value || isNullOrUndefined(value.calculatorId)
-        ? this.#getDefaultValue(this.calculators[0])
-        : {
-            ...this.#getDefaultValue(this.#getCalculator(value.calculatorId)),
-            ...value,
-          };
+    const isValueEmpty = !value || isNullOrUndefined(value.calculatorId);
+    const valueOrDefault = isValueEmpty
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts:549]: The logic of this condition didn't change; I just tweaked it a bit for readability.

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2738/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2738/storybooks/ag-grid-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2738/storybooks/datetime-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2738/storybooks/lists-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2738/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2738/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2738/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2738/playground/)

File: libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.spec.ts [hunk 2]
```diff
@@ -702,6 +702,22 @@ describe('Date range picker', function () {
...
+    expect(component.reactiveForm.value).toEqual({
+      dateRange: { calculatorId: 5 },
+    });
+
+    selectCalculator(SkyDateRangeCalculatorId.AnyTime);
+    detectChanges();
+
+    expect(component.reactiveForm.value).toEqual({
+      dateRange: { calculatorId: 0, startDate: null, endDate: null },
+    });
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.spec.ts:725]: could we add a unit test switching between calculators with no datepickers but going to one that does have startDate and endDate set? (eg Last Month)

Comment (Blackbaud-SteveBrush) [reply libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.spec.ts:725]: Done.

**Review by Blackbaud-SandhyaRajasabeson — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/2740

**Pull request merged**